contracts/erc20-gateway/ERC20Gateway.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../proxies/MasterCopyNonUpgradable.sol";
+import "../message-bus/MessageBus.sol";
+
+contract ERC20Gateway is MasterCopyNonUpgradable, MessageBus {
+
+    /* Constants */
+
+    /** Storage offset of message outbox. */
+    uint8 constant public OUTBOX_OFFSET = uint8(1);
+
+    /** Storage offset of message inbox. */
+    uint8 constant public INBOX_OFFSET = uint8(4);
+
+
+    /* External Functions */
+
+    /**
+     * @notice Setup function for ERC20 gateway contract.
+     *
+     * @dev - Function can only be called once is ensured by setup function of
+     *        message outbox and message inbox.
+     *      - Validations for input parameters are done in message outbox and
+     *        message inbox setup method.
+     *
+     * @param _metachainId Metachain Id.
+     * @param _erc20Cogateway Address of ERC20 Cogateway contract.
+     * @param _stateRootProvider State root provider contract address.
+     * @param _maxStorageRootItems Maximum number of storage roots stored.
+     * @param _outboxStorageIndex Outbox storage index of ERC20 Cogateway.
+     *
+     * \post Setup message outbox and updates outboundChannelIdentifier storage
```

Comment:
`\pre This function can only be called once.`
Add a precondition, it can be removed from the `@dev` and can be used here as a precondition

Reply:
Added this as `pre` condition. 
I think we should keep @dev since it also explains how it's made sure that this function is called only once.

Reply:
@abhayks1 I think we should not duplicate documentation in any way. Keeping documentation sync with implementation is already hard, having it twice in any way will add into it. Hence, we should not duplicate what we already have defined in `\pre` and `\post` conditions in @dev tag documentation.

Reply:
Also, we should either specify all `\pre` and `\post` conditions in external/public function or if we delegate some of their validations to other public/internal functions clearly refer them with `\ref` tag. (I would assume private functions do no validations, or basically assert if they do). @deepesh-kn @abhayks1 if you agree, please, let me know here and I can update it in developer guidelines.

Reply:
@pgev agree with not duplicating documentation. My only point was regarding the extra detail which is updated with latest change.
cc: @jayeshbairagi @deepesh-kn for the guideline update.

Reply:
@pgev, sure we can update this in the developer guidelines.
Making note of this for discussion. Going to merge this.